### PR TITLE
Тюрмина А.Н. 381806-3 ЛР3

### PR DESCRIPTION
**Pull request by sashatyurmina**
### Pull Request Readiness Checklist

See details at [code review]( https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/main/lab-guide/lab-1-setup.md#%D1%80%D0%B5%D1%86%D0%B5%D0%BD%D0%B7%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

- [ ] Code is well documented and more information can be seen on the [website](https://devtools-course-practice-2021.readthedocs.io/en/latest/modules.html) (recommended)
- [ ] Make sure that the testing has passed successfully on [GitHub Actions](https://github.com/UNN-ITMM-Software/devtools-course-practice/actions)
- [ ] Get two approve from your fellow students
- [ ] Take a review from one of the administrators

### CODE REVIEW

**Comment by sashatyurmina** [comment]
@dasha-ludina @shekotilova-yuliya

File: modules/leftist-heap/application/application.cpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2021 Tyurmina Alexandra
+
+#include <iostream>
+#include <string>
+#include "include/leftist_heap_app.h"
```
Comment (dasha-ludina) [inline modules/leftist-heap/application/application.cpp:5]: Необходимо добавить пустую строку

File: modules/leftist-heap/include/leftist_heap_app.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2021 Alexandra Tyurmina
+
```
Comment (dasha-ludina) [inline modules/leftist-heap/include/leftist_heap_app.h:2]:
```suggestion
// Copyright 2021 Tyurmina Alexandra

```

File: modules/leftist-heap/include/leftist_heap_app.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#ifndef MODULES_LEFTIST_HEAP_INCLUDE_LEFTIST_HEAP_APP_H_
+#define MODULES_LEFTIST_HEAP_INCLUDE_LEFTIST_HEAP_APP_H_
+
+#include <string>
+#include "include/leftist_heap.h"
```
Comment (dasha-ludina) [inline modules/leftist-heap/include/leftist_heap_app.h:7]: Необходимо добавить пустую строку

File: modules/leftist-heap/include/leftist_heap_app.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+class HeapSort {
+ public:
+  std::string operator()(int argc, const char** argv);
+ private:
+  std::string message_;
+  void Info(const char* appname, const char* message = "");
+  bool CheckArguments(int argc, const char** argv);
```
Comment (dasha-ludina) [inline modules/leftist-heap/include/leftist_heap_app.h:15]: Лучше поменять местами private и public

```suggestion
 private:
  std::string message_;
  void Info(const char* appname, const char* message = "");
  bool CheckArguments(int argc, const char** argv);
  
 public:
  std::string operator()(int argc, const char** argv);
```

File: modules/leftist-heap/src/leftist_heap_app.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Tyurmina Alexandra
+
+#include "include/leftist_heap_app.h"
+#include <string>
+#include <cstring>
```
Comment (dasha-ludina) [inline modules/leftist-heap/src/leftist_heap_app.cpp:5]:
```suggestion
#include <string>
#include <cstring>

#include "include/leftist_heap_app.h"
```

File: modules/leftist-heap/test/test_leftist_heap_app.cpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 Tyurmina Alexandra
+
+#include <vector>
+#include <string>
+#include "include/leftist_heap_app.h"
```
Comment (dasha-ludina) [inline modules/leftist-heap/test/test_leftist_heap_app.cpp:5]: Необходимо добавить пустую строку

**Review by dasha-ludina — APPROVED** [review]

**Review by shekotilova-yuliya — COMMENTED** [review]
все отлично!

**Review by shekotilova-yuliya — APPROVED** [review]

**Review by WerWebWer — APPROVED** [review]

**Pull request merged**